oak/server/rust/oak_runtime/src/lib.rs

```diff
@@ -568,27 +568,21 @@ impl OakNode {
 }
 
 /// Expected type for the main entrypoint to a Node under test.
-pub type NodeMain = fn(Handle) -> i32;
+pub type NodeMain = fn(Handle) -> Result<(), oak::OakStatus>;
```

Comment:
Hmm, I think this change is going in the wrong direction &ndash; it makes the Rust runtime more test-specific (because it assumes access to Rust types) and less amenable to "real" use (where Node entrypoints are going to have signature `extern "C" (u64)  -> i32`).

(Apologies, didn't notice this before)

Reply:
The alternative could be to use "rich" types internally for convenience in error handling (e.g. when defining `log_node_main` below), but only use  FFI types for invoking them from `oak_runtime` and `oak_tests`, using some adaptor functions. Let me try something and send you a PR.